doc/data/messages/a/arguments-differ/good.py

```diff
@@ -0,0 +1,23 @@
+class Rectangle:
+    def __init__(self, length):
+        self.length = length
+
+
+class Square(Rectangle):
+    def __init__(self, length, width=None):
+        self.length = length
+        self.width = width
```

Comment:
```suggestion
"""
'argument-differ' denotes of a a high level design issue regarding the L in SOLID
(Liskov Substitution Principle), so there is no easy fix or a single possible fix.
You might want to read this:
https://softwareengineering.stackexchange.com/a/238184/224126
In short you could:
- Make the base class immutable
- Remove inheritance completely
- Add default arguments (It makes a square behave like a rectangle but did you check everything ? What happen in the square if you want to increase the area by modifying the width of the rectangle ?)
"""

from dataclasses import dataclass


@dataclass(frozen=True)
class Rectangle:
    length: float
    width: float


class Square(Rectangle):
    def __init__(self, length, width=None):
        assert (
            width is None or length == width
        ), "Length is not equal to width in a square."
        super(Square, self).__init__(length, length)


square = Square(5)
rectangle = Rectangle(4, 6)       
```

Reply:
The docstring should probably go in `details.rst`. I'll have look at this next week!